src/detext/run_detext.py

```diff
@@ -2,275 +2,148 @@
 Overall pipeline to train the model.  It parses arguments, and trains a DeText model.
 """
 
-import argparse
 import logging
 import os
 import sys
 import time
+from typing import NamedTuple, List
 
 import tensorflow as tf
 import tensorflow_ranking as tfr
+from arg_suite.arg_suite import arg_suite
 from detext.train import train
 from detext.train.data_fn import input_fn
 from detext.utils import misc_utils, logger, executor_utils
 
 
-def add_arguments(parser):
-    """Build ArgumentParser."""
-    parser.register("type", "bool", lambda v: v.lower() == "true")
-
+@arg_suite
+class Args(NamedTuple):
     # network
-    parser.add_argument("--ftr_ext", choices=['cnn', 'bert', 'lstm_lm', 'lstm'], help="NLP feature extraction module.")
-    parser.add_argument("--num_units", type=int, default=128, help="word embedding size.")
-    parser.add_argument("--num_units_for_id_ftr", type=int, default=128, help="id feature embedding size.")
-    parser.add_argument("--sp_emb_size", type=int, default=1, help="Embedding size of sparse features")
-    parser.add_argument("--num_hidden", type=str, default='0', help="hidden size.")
-    parser.add_argument("--num_wide", type=int, default=0, help="number of wide features per doc.")
-    parser.add_argument("--num_wide_sp", type=int, default=None, help="number of sparse wide features per doc")
-    parser.add_argument("--use_deep", type=str2bool, default=True, help="Whether to use deep features.")
-    parser.add_argument("--elem_rescale", type=str2bool, default=True,
-                        help="Whether to perform elementwise rescaling.")
-    parser.add_argument("--use_doc_projection", type=str2bool, default=False,
-                        help="whether to project multiple doc features to 1 vector.")
-    parser.add_argument("--use_usr_projection", type=str2bool, default=False,
-                        help="whether to project multiple usr features to 1 vector.")
+    ftr_ext: str  # NLP feature extraction module.
+    _ftr_ext = {'choices': ['cnn', 'bert', 'lstm_lm', 'lstm']}
+    num_units: int = 128  # word embedding size.
+    num_units_for_id_ftr: int = 128  # id feature embedding size.
+    sp_emb_size: int = 1  # Embedding size of sparse features
+    num_hidden: List[int] = 0  # hidden size.
+    num_wide: int = 0  # number of wide features per doc.
+    num_wide_sp: int = None  # number of sparse wide features per doc
+    use_deep: bool = True  # Whether to use deep features.
```

Comment:
Will lower case input such as `false` and `true` treated as `False` and `True` correspondingly?

Reply:
It's strictly 'True or False' as in the help. otherwise, an error will be raised.

Reply:
Can we support the lower case input? It's foreseeable that we will have a lot of job failures because of this change. It's easy to make mistakes without this casing tolerance.

Reply:
yes, we can customize this for backward compatibility, but for future ones, I'd suggest sticking to the default.